cachi2/core/package_managers/rpm/main.py

```diff
@@ -0,0 +1,204 @@
+        raw_content = yaml.safe_load(f)
+
+        # Create a list of format handlers and add handler for every supported format
+        # Lockfile processor will use the first handler that matches the input format
+        lockfile_format_processor = LockfileFormat(raw_content)
+        lockfile_format_processor.add_handler(RedhatRpmsLock)
+        lockfile_format_processor.process_formats()
+
+        package_dir = request.output_dir.join_within_root(DEFAULT_PACKAGE_DIR)
+        handler = lockfile_format_processor.matched_handler
+        handler.download(package_dir)
+        handler.verify_downloaded()
+        files_metadata = handler.metadata
+        _generate_sbom_components(components, files_metadata)
+
+    return RequestOutput.from_obj_list(
+        components=components,
+        environment_variables=_generate_environment_variables(DEFAULT_PACKAGE_DIR),
+        project_files=_generate_repofiles(request.output_dir),
+    )
+
+
+def _generate_sbom_components(components: list[Component], files_metadata: dict) -> None:
+    """Fill the compoenent list with the package records."""
+    for file_path, file_metadata in files_metadata.items():
+        # iterate only through package metadata (skip sources metadata)
+        if not file_metadata["package"]:
+            continue
+        # get (optional) 'epoch' - can't get it via rpmfile.headers
+        rpm_args = [
+            "-q",
+            "--queryformat",
+            "%|EPOCH?{%{EPOCH}}:{}|",  # return "" when epoch is not set instead of "(None)"
+            file_path,
+        ]
+        epoch = run_cmd(cmd=["rpm", *rpm_args], params={})
+
+        # NOTE: add license into SBOM?
+        rpm_args = [
+            "-q",
+            "--queryformat",
+            "%{LICENSE}",
+            file_path,
+        ]
+        license = run_cmd(cmd=["rpm", *rpm_args], params={})
+
+        with rpmfile.open(file_path) as rpm:
+            vendor = rpm.headers.get("vendor", b"").decode()
+            vendor = quote(vendor.lower())
+
+            name = rpm.headers.get("name", b"").decode()
+            version = rpm.headers.get("version", b"").decode()
+            release = rpm.headers.get("release", b"").decode()
+            arch = rpm.headers.get("arch", b"").decode()
+            download_url = quote(file_metadata["url"])
+
+            # https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#rpm
+            # https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#known-qualifiers-keyvalue-pairsa
+            purl = (
+                f"pkg:rpm{'/' if vendor else ''}{vendor}/{name}@{version}{release}"
+                f"?arch={arch}{'&epoch=' if epoch else ''}{epoch}&download_url={download_url}"
+            )
+
+            components.append(
+                Component(
+                    name=name,
+                    version=version,
+                    purl=purl,
+                    # NOTE: add checksum into SBOM?
```

Comment:
For now, we're not adding checksums in the SBOM in any of the supporte package managers, but we are reporting cases in which the checksum was missing from the lockfile. I believe we should do the same for RPMs.

Here's an example of how it looks like in the SBOM: https://github.com/containerbuildsystem/cachi2/blob/main/tests/integration/test_data/gomod_multiple_modules_missing_checksums/bom.json#L87-L101

Reply:
OK, what should be the Property value in this case? Is file name (rpm name) enough?